nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1426 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import shutil
+import warnings
+from typing import Dict, List, Tuple
+
+import numpy as np
+import soundfile as sf
+import torch
+from scipy.signal import convolve
+from scipy.signal.windows import cosine, hamming, hann
+from scipy.stats import halfnorm
+from tqdm import trange
+
+from nemo.collections.asr.parts.utils.manifest_utils import (
+    create_manifest,
+    create_segment_manifest,
+    read_manifest,
+    write_ctm,
+    write_manifest,
+    write_text,
+)
+from nemo.collections.asr.parts.utils.speaker_utils import labels_to_rttmfile
+from nemo.utils import logging
+
+try:
+    import pyroomacoustics as pra
+    from pyroomacoustics.directivities import CardioidFamily, DirectionVector, DirectivityPattern
+
+    PRA = True
+except ImportError:
+    PRA = False
+try:
+    from gpuRIR import att2t_SabineEstimator, beta_SabineEstimation, simulateRIR, t2n
+
+    GPURIR = True
+except ImportError:
+    GPURIR = False
+
+
+class MultiSpeakerSimulator(object):
+    """
+    Multispeaker Audio Session Simulator.
+    Simulates multispeaker audio sessions using single-speaker audio files and corresponding word alignments.
+
+    Args:
+        cfg: OmegaConf configuration loaded from yaml file.
+
+    Parameters:
```

Comment:
Please make line-breaks if the sentences are too long. 
Also, make sure it starts with either lower case or upper case.